### PR TITLE
nimble/mesh: Generate new public key on each provisioning session

### PR DESCRIPTION
**Pull request by michal-narajowski**
Mesh Profile Spec v1.0.1 | Section 5.4.2.3:
`If the public key was not available using an OOB technology,
then the public keys are exchanged between the Provisioner
and the unprovisioned device. For each exchange, a new key pair shall be
generated by the Provisioner and the unprovisioned device.`

This allows passing MESH/NODE/PROV/BV-12-C.

### CODE REVIEW

**Comment by rymanluk** [comment]
@jhedberg, mentioned testcase is enabled not only for Mesh v1.0.1 but also for Erratum 10395 which is Mandatory for Mesh v1.0

**Comment by michal-narajowski** [comment]
Fixed a conflict.

**Comment by michal-narajowski** [comment]
@jhedberg FYI

File: nimble/host/mesh/src/prov.c [hunk 1]
```diff
@@ -249,8 +257,10 @@ static void reset_link(void)
 #endif
 	link.rx.prev_id = XACT_NVAL;
 
-	if (bt_pub_key_get()) {
-		atomic_set_bit(link.flags, LOCAL_PUB_KEY);
+	err = bt_pub_key_gen(&pub_key_cb);
```
Comment (jhedberg) [inline nimble/host/mesh/src/prov.c]: At least in Zephyr `bt_pub_key_gen()` stores the callback persistently in a linked list. As long as we call this from `init()` there's no problem (it's only called once) but when you call it from `reset()` you'll end up with a corrupted linked list. I'm currently looking at how/if we should change our `bt_pub_key_gen()` implementation to clear the list when generation is complete (i.e. the callbacks wouldn't be persistently stored)

**Comment by michal-narajowski** [comment]
This PR has been superseded by https://github.com/apache/mynewt-nimble/pull/506

**Pull request closed without merging**